yt_dlp/extractor/vanillo.py

```diff
@@ -0,0 +1,386 @@
+                break
+            # For each comment, download replies (if any)
+            for comment in comments:
+                transformed = {
+                    'id': comment.get('id'),
+                    'author': comment.get('profile', {}).get('username'),
+                    'author_id': comment.get('profile', {}).get('id'),
+                    'text': comment.get('text'),
+                    'timestamp': parse_iso8601(comment.get('createdAt')),
+                    'replies': self._get_replies(comment.get('id')),
+                }
+                all_comments.append(transformed)
+            page_key = data.get('nextPageKey')
+            if not page_key:
+                break
+        return all_comments
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        # 1) Retrieve video info (metadata)
+        video_info_url = f'https://api.vanillo.tv/v1/videos/{video_id}?groups=uploader,profile.full'
+        try:
+            video_info = self._download_json(video_info_url, video_id, note='Downloading video info')
+        except ExtractorError as e:
+            # Try to get an HTTP code from the error cause or message
+            http_code = getattr(e.cause, 'code', None)
+            if http_code is None and 'HTTP Error 404' in str(e):
+                http_code = 404
+            if http_code == 404:
+                self.raise_login_required(
+                    'Video not found or is private. '  # maybe use report_login here?
+                    'Sign in if you have access to this video. Use --cookies or --cookies-from-browser option',
+                    method=None,
+                )
+            elif http_code == 403:
+                raise ExtractorError(
+                    'Your Internet provider is likely blocked. Try again with another ISP or use VPN',
+                    expected=True,
+                )
+            raise
+
+        if video_info.get('status') != 'success':
+            raise ExtractorError('Video info API returned an error', expected=True)
+        data = video_info.get('data', {})
+        title = data.get('title') or video_id
+        description = data.get('description')
+        thumbnail = data.get('thumbnail')
+
+        uploader = data.get('uploader', {})
+        uploader_url = uploader.get('url')
+
+        # 2) Fix the ISO8601 date to remove leftover data
+        upload_date_raw = data.get('publishedAt')
+        upload_date = None
+        if upload_date_raw:
+            # Remove fractional seconds and any extra data after 'Z'
+            upload_date_raw = re.sub(r'\.\d+', '', upload_date_raw)
+            upload_date_raw = re.sub(r'Z.*$', 'Z', upload_date_raw)
+            try:
+                parsed_date = datetime.datetime.fromisoformat(upload_date_raw.replace('Z', '+00:00'))
+                upload_date = parsed_date.strftime('%Y%m%d')
+            except ValueError:
+                pass
+
+        duration = data.get('duration')
```

Comment:
Don't use `.get()` chains like this; use `traverse_obj` (`from ..utils.traversal import traverse_obj`).

Use `parse_iso8601` to extract the `timestamp` instead of manually mangling it into an `upload_date`

And you can inline most/all of this in the return